detox/src/devices/Device.test.js

```diff
@@ -478,7 +478,7 @@ describe('Device', () => {
 
       await device.installApp('newAppPath');
 
-      expect(driverMock.driver.installApp).toHaveBeenCalledWith(device._deviceId, 'newAppPath', undefined);
+      expect(driverMock.driver.installApp).toHaveBeenCalledWith(device._deviceId, 'newAppPath', device._deviceConfig.testBinaryPath);
```

Comment:
Why did this change from `undefined`  to `testBinaryPath` ?

Reply:
Previously this expectation was relying on implementation details of the `Device` class. Specifically, it knew that testBinaryPath was not passed along to `newAppPath` if `prepare` has not yet been called. That's because previously, the binary path instance properties were set on the instance during `prepare` after doing the absolute path logic I have pushed down into the driver layer.

That seemed a bit odd to me, since in all real uses of Device (as far as I'm aware) prepare will always be called before using the other methods. This method expects the testBinaryPath as the last argument, it just previously happened to not be defined yet